File: 1482. Minimum Number of Days to Make m Bouquets.go
func minDays(bloomDay []int, m int, k int) int {
    left, right, answer := 1000000000, 0, -1
    for _, i := range bloomDay {
        if left > i { left = i }
        if right < i { right = i }
    }
    for left <= right {
        mid := (left + right) / 2
        if canDo(bloomDay, mid, m, k) {
            answer = mid
            right = mid - 1
        } else {
            left = mid + 1
        }
    }
    return answer
}

func canDo(arr []int, val, m, k int) bool {
    count := 0
    for _, i := range arr {
        if i <= val {
            count += 1
        } else {
            count = 0
        }
        if count == k {
            count = 0
            m -= 1
        }
    }
    return m <= 0
}

File: java/1482. Minimum Number of Days to Make m Bouquets.java
class Solution {
    public int minDays(int[] bloomDay, int m, int k) {
        long right = Integer.MIN_VALUE;
        long left = Integer.MAX_VALUE;
        for (int i : bloomDay) {
            right = Math.max(right, i);
            left = Math.min(left, i);
        }
        int answer = -1;
        while (left <= right) {
            long mid = (left + right)/2;
            if (canDo(bloomDay, m, k, (int)mid)) {
                answer = (int)mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return answer;
    }
    private boolean canDo(int[] n, int m, int k, int currentDay) {
        int count = 0;
        int currentWindow = 0;
        for (int i : n) {
            if (i <= currentDay) {
                currentWindow++;
            } else {
                currentWindow = 0;
            }
            if (currentWindow == k) {
                count++;
                currentWindow = 0;
            }
        }
        return count >= m;
    }
}
